### ISSUE TITLE
Stuggling to get Markdown working

### ISSUE THREAD

**Issue by khrome83** [littletof/prettyBenching#10]
This is what I am getting when I try to run it.

I am using a small sample of your code, and it does not seem to run.

```bash
error: Uncaught TypeError: Cannot read property 'replace' of undefined
    ? `${code.open}${str.replace(code.regexp, code.open)}${code.close}`
                         ^
    at run (colors.ts:54:26)
    at red (colors.ts:95:10)
    at benchmarks.ts:46:19
```

```ts
  runBenchmarks({ silent: true, skip: /_long/ })
    .then(
      prettyBenchmarkDown(console.log, {
        title: "Example Markdown",
        description: (runResult: BenchmarkRunResult) =>
          `This markdown was generated with the use of \`prettyBenchmarkDown\`.\nIf you use a function for the \`description\` or \`afterTables\`, you can process the results here as well: \n\n > In this benchmark ${runResult.results.length} benches were run, ${runResult.filtered} were filtered.`,
        afterTables:
          "\n---\n\nThis is the `afterTables`. This behaves the same as `description`, it just puts this at the bottom of the markdown.\nHere its defined with a simple string.\n\nCheck out the Github Action, which comments a markdown like this on PRs: $link",
        groups: [
          {
            include: /array/,
            name: "Default columns and dynamic text",
            description:
              "This is a group's `description`.\nHere you can see what the default columns are, and how you can use a `function` as `description` or `afterTable` inside a group",
            afterTable: (
              gr: BenchmarkResult[],
              g: GroupDefinition,
              rr: BenchmarkRunResult
            ) =>
              `This is a group's \`afterTable\`.\nHere you can access eg. the group name: \`${g.name}\`, benchmarks in this group: \`${gr.length}\` of them here, or the whole \`BenchmarkRunResult\`: \`${rr.results.length}\` benchmarks total`,
            columns: [...defaultColumns],
          },
        ],
      })
    )
    .catch((e: BenchmarkRunError) => {
      console.log(red(e.benchmarkName as string));
      console.error(red(e.stack as string));
    });
```

**Comment by khrome83** [comment]
Apologies @littletof you can see the output here

https://github.com/use-seedling/seedling/pull/3

**Comment by littletof** [comment]
Will take a look at it tomorrow. 

There is no coloring inside `prettyBenchmarkDown` if I'm correct, so is comes from somewhere else.
I got errors like this, when trying to use color functions on `undefined` vaules , so maybe an error is thrown, where `e.benchmarkName` is `undefined`.

Can you check if this is the case please?

**Comment by littletof** [comment]
You got the error, because in the `afterTable` function the parameter `gr` was `undefined`, because the group was empty. It was empty because `include` did not match any of your benchmark names.

This should be fixed with #11, and be in [`v0.2.1`](https://github.com/littletof/prettyBenching/pull/12), where these functions will recieve an empty array for `gr`, if the same case happens and the `gr` would be `undefined` otherwise.

You got that specific error, because your `catch` branch caught an error, on which `benchmarkName` is `undefined`, and Deno's coloring functions fail, when you provide `undefined` to them. 😕
It would be better to remove that line, or at least put it below `e.stack`.
( I know this catch was in examples in this repo, but since I tried to remove them for this exact reason. If you know any that is still present, please point to it )
